jdatetime/__init__.py

```diff
@@ -332,6 +332,20 @@ def fromtimestamp(timestamp):
         d = py_datetime.date.fromtimestamp(timestamp)
         (y, m, d) = GregorianToJalali(d.year, d.month, d.day).getJalaliList()
         return date(y, m, d)
+    
+    @staticmethod
+    def fromisoformat(date_string:str):
+        """
+        Convert an ISO 8601 formatted string to a date
+        :param date_string: ISO 8601 formatted string
+        """
+        date_list = list(map(int, date_string.split('-')))
+        if len(date_list) == 3:
+            y, m, d = date_list
+            return date(y, m, d)
+        else:
+            raise ValueError('Invalid iso format date string')
```

Comment:
Please include the `value` in the error message to be consistent with Python.

```
>>> date.fromisoformat('wrong date string')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: Invalid isoformat string: 'wrong date string'
```

Reply:
ok tnx